### PR TITLE
Added Sine function algorithm

### PR DESCRIPTION
**Pull request by XZANATOL**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

- [x] Yes

### Description

Added the algorithm for the Sine function as assigned in #2906 Issue

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

Issue number: #2906

### CODE REVIEW

**Comment by XZANATOL** [comment]
Most of the failed checks considering the Python action failed to pass becuase of defining same names of variables in different scopes, but each has its own job. Shall I resubmit the Script using other names?

**Comment by ankitaggarwal23** [comment]
You can ignore it, there is some glitch at deepsource's end.

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]
Few changes: 

- Add Multi-Line Comments 
- Add a beginner-friendly description at the top
- Use proper code Format 
- Remove extra lines of code and spaces 
- Add sample I/O at the end of the Code
- Add Time and Space Complexity at the end of the Code

**Comment by XZANATOL** [comment]
@ankitaggarwal23 Requested changes has been added.

**Comment by XZANATOL** [comment]
@ankitaggarwal23 any updates?

File: Python/math/Sine_Function.py [hunk 1]
```diff
@@ -0,0 +1,123 @@
+#Sine Algorithm
+#Written by XZANATOL
+#Algorithm is a contribution to Neo-Algo at https://github.com/TesseractCoding/NeoAlgo
```
Comment (ankitaggarwal23) [inline Python/math/Sine_Function.py]: Kindly Remove this line.

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]
- Remove Extra Spaces or new line(if any)

**Comment by XZANATOL** [comment]
@ankitaggarwal23 Done

**Review by ankitaggarwal23 — APPROVED** [review]

**Pull request merged**